File: backend/src/main/kotlin/ru/itmo/db/coursework/api/mapping/UserDtoMapper.kt
package ru.itmo.db.coursework.api.mapping

import org.springframework.stereotype.Service
import ru.itmo.db.coursework.api.dto.UserDto
import ru.itmo.db.coursework.model.User

@Service
class UserDtoMapper {
    fun fromDto(dto: UserDto, passwordHash: String) = with(dto, {
        User(
            id = id,
            login = login,
            name = name,
            passwordHash = passwordHash,
            money = money
        )
    })

    fun toDto(model: User) = with(model, {
        UserDto(
            id = id,
            login = login,
            name = name,
            money = money
        )
    })
}

File: frontend/src/api/mapping/UserDtoMapper.ts
import UserDto from "$src/api/dto/UserDto";
import User from "$src/game/model/User";

export default class UserDtoMapper {
  public static fromDto(dto: UserDto) {
    return new User(dto.id, dto.login, dto.name, dto.money)
  }

  public static toDto(model: User) {
    return new UserDto(model.id, model.login, model.name, model.money)
  }
}